pyinteraph/path_analysis.py

```diff
@@ -204,113 +214,205 @@ def get_all_shortest_paths(graph, res_id, res_space):
+    # Rocket_r
+    cb = sns.color_palette("rocket_r")
+    cmap_e = LinearSegmentedColormap.from_list('cb', cb , N=100)
+    # Remove border
+    fig, ax = plt.subplots()
+    ax.axis('off')
+    # Draw non hubs
+    nx.draw_networkx_nodes(graph, pos, node_list = list(non_hubs),
+                                       node_size = 900,
+                                       node_color = 'white',
+                                       edgecolors = 'black',
+                                       label = list(non_hubs_deg))
+    # Draw hubs
+    if unique_hubs == 1:
+        # COlor the unique hub gray
+        nx.draw_networkx_nodes(graph, pos, nodelist = list(hubs),
+                                        node_size = 900,
+                                        node_color = 'gray',
+                                        edgecolors = 'black')
+    elif unique_hubs > 1:
+        # Use colormap if more than 1 unique hubs
+        nx.draw_networkx_nodes(graph, pos, nodelist = list(hubs),
+                                        node_size = 900,
+                                        node_color = list(hubs_deg),
+                                        cmap = cmap_n,
+                                        edgecolors = 'black',
+                                        label = list(hubs_deg))
+    # Draw edges
+    nx.draw_networkx_edges(graph, pos, edge_color = weights, 
+                                       edge_cmap = cmap_e,
+                                       edge_vmin = 0, 
+                                       edge_vmax= 1,
+                                       width = 3)
+    # Draw labels
+    nx.draw_networkx_labels(graph, pos, font_size=9, font_color ='black')
+    # Add color bar
+    # Resize colorbar
+    divider = make_axes_locatable(ax)
+    cax_e = divider.append_axes("right", size="5%", pad=0.2)
+    # Edge color bar
+    sm_e = plt.cm.ScalarMappable(cmap=cmap_e, norm=plt.Normalize(vmin=0, 
+                                                              vmax=1))
+    cbar_e = plt.colorbar(sm_e, cax_e)
+    cbar_e.set_label('Relative recurrence')
+    # Node color bar (gray scale)
+    if unique_hubs > 1:
+        cax_n = divider.append_axes("bottom", size="5%", pad=0.2)
+        lo = min(hubs_deg)
+        hi = max(hubs_deg)
+        sm_n = plt.cm.ScalarMappable(cmap=cmap_n, 
+                                     norm=plt.Normalize(vmin=lo, 
+                                                        vmax=hi))
+        # Find tick spacing
+        diff = (hi-lo)/hi
+        start = lo + diff/2
+        end = hi - diff/2 + 1
+        # Add ticks
+        cbar_n = plt.colorbar(sm_n, cax_n, 
+                            orientation = 'horizontal',
+                            ticks = np.arange(start, end, diff))
+        # Add tick labels
+        cbar_n.ax.set_xticklabels(range(lo, hi + 1))
+        cbar_n.set_label('Node Degree')
+    # Save figure
+    plt.savefig(fname, dpi = 100)
```

Comment:
You can also pass the DPI argument to the function instead of having it hard-coded  :) 

Have a look at the `bbox_inches` argument of `savefig` and consider setting it to `tight_layout`, since it can avoid some unwanted extra spaces between elements of a plot.